File: Common/InMotionGIT.Common - Copy/DataAcceess/WhereStatements/DateCondition.vb
Namespace DataAccess.WhereStatements

    Public Class DateCondition(Of t)

        Protected Friend ReadOnly fieldName As String
        Protected Friend ReadOnly ConnectionName As String
        Protected Friend ReadOnly tableType As IWhereStatement

        Public Sub New(name As String, base As IWhereStatement)
            fieldName = name
            tableType = base
        End Sub

        Public Sub New(name As String, connectionName As String, base As IWhereStatement)
            fieldName = name
            tableType = base
            Me.ConnectionName = connectionName
        End Sub

        Public Function EqualTo(value As Date) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} = '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value))
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function NotEqualTo(value As Date) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} <> '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value))
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function GreaterThan(value As Date) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} > '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value))
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function LessThan(value As Date) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} < '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value))
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function GreaterThanEqualTo(value As Date) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} >= '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value))
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function LessThanEqualTo(value As Date) As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} <= '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value))
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function IsNull() As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} IS NULL", fieldName)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function IsNotNull() As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} IS NOT NULL", fieldName)
            Return New LogicalOperator(Of t)(tableType)
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/DataAcceess/WhereStatements/DateCondition.cs
using System;

namespace InMotionGIT.Common.DataAccess.WhereStatements
{

    public class DateCondition<t>
    {

        protected internal readonly string fieldName;
        protected internal readonly string ConnectionName;
        protected internal readonly IWhereStatement tableType;

        public DateCondition(string name, IWhereStatement @base)
        {
            fieldName = name;
            tableType = @base;
        }

        public DateCondition(string name, string connectionName, IWhereStatement @base)
        {
            fieldName = name;
            tableType = @base;
            ConnectionName = connectionName;
        }

        public LogicalOperator<t> EqualTo(DateTime value)
        {
            tableType.command += string.Format("{0} = '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value));
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> NotEqualTo(DateTime value)
        {
            tableType.command += string.Format("{0} <> '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value));
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> GreaterThan(DateTime value)
        {
            tableType.command += string.Format("{0} > '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value));
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> LessThan(DateTime value)
        {
            tableType.command += string.Format("{0} < '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value));
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> GreaterThanEqualTo(DateTime value)
        {
            tableType.command += string.Format("{0} >= '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value));
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> LessThanEqualTo(DateTime value)
        {
            tableType.command += string.Format("{0} <= '{1}'", fieldName, Helpers.DataAccessLayer.DateValueWithFormat(ConnectionName, value));
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> IsNull()
        {
            tableType.command += string.Format("{0} IS NULL", fieldName);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> IsNotNull()
        {
            tableType.command += string.Format("{0} IS NOT NULL", fieldName);
            return new LogicalOperator<t>(tableType);
        }

    }

}